Docs: clarify where counter variables start (#45247)
rails/rails: guides/source/layouts_and_rendering.md

In this case, the partial will have access to a local variable `title` with the value "Products Page".

#### Counter Variables

> Rails also makes a counter variable available within a partial called by the collection, named after the title of the partial followed by `_counter`. For example, when rendering a collection `@products` the partial `_product.html.erb` can access the variable `product_counter` which indexes the number of times it has been rendered within the enclosing view, starting with a value of `0` on the first render.

Reviewer: Some of these sentences are pretty long.
What do you think about cutting them up a bit?
Maybe something like:
```suggestion
Rails also makes a counter variable available within a partial called by the collection. The variable is named after the title of the partial followed by `_counter`. For example, when rendering a collection `@products` the partial `_product.html.erb` can access the variable `product_counter`. The variable indexes the number of times the partial has been rendered within the enclosing view, starting with a value of `0` on the first render.
```

In this case, the partial will have access to a local variable `title` with the value "Products Page".

> TIP: Rails also makes a counter variable available within a partial called by the collection, named after the title of the partial followed by `_counter`. For example, when rendering a collection `@products` the partial `_product.html.erb` can access the variable `product_counter` which indexes the number of times it has been rendered within the enclosing view, starting with a value of `0` on the first render. Note that it also applies for when the partial name was changed by using the `as:` option. For example, the counter variable for the code above would be `item_counter`.

Reviewer: Hey @ghiculescu 👋 .
This whole TIP is a bit difficult to understand/visualize.
What do you think about adding an example (and not render it as a TIP)?
